### PR TITLE
Virt-api, Arm64: Add error message for watchdog and sound device on Arm64

### LINKED ISSUE #8916: Kubevirt Device Verification on Arm64
Devices | Description | Status on Arm64
-- | -- | --
DisableHotplug |   | supported
Disks | sata/ virtio busmake | support virtio bus
Watchdog | i6300esb | not support
UseVirtioTransitional | virtio-transitional | supported
Interfaces | e1000/ virtio-net-device | support virtio-net-device
Inputs | tablet virtio/usb bus | support
AutoattachPodInterface | connect to /net/tun (devices.kubevirt.io/tun) | support
AutoattachGraphicsDevice | create a virtio-gpu device / vga device | support virtio-gpu
AutoattachMemBalloon | virtio-balloon-pci-non-transitional | support
AutoattachInputDevice | auto add tablet | support
Rng | virtio-rng-pci-non-transitional host:/dev/urandom | support
BlockMultiQueue | "driver":"virtio-blk-pci-non-transitional","num-queues":$cpu_number | support
NetworkInterfaceMultiQueue | -netdev tap,fds=21:23:24:25,vhost=on,vhostfds=26:27:28:29,id=hostua-default#fd number equals to queue number | supported
GPUs |   | not verified
Filesystems | virtiofs, vhost-user-fs-pci, need to enable featuregate: ExperimentalVirtiofsSupport | supported
ClientPassthrough | https://www.linaro.org/blog/kvm-pciemsi-passthrough-armarm64/on x86_64, iommu need to be enabled | not verified
Sound | ich9/ ac97 | not support
TPM | tpm-tis-devicehttps://qemu.readthedocs.io/en/latest/specs/tpm.html | supported
Sriov | vfio-pci | not verified

Summary: 
Watchdog device, Sound device are not supported on Arm64
Beyond them, secure boot is also not support on Arm64. 
SRIOV, GPU and ClientPassthrough still need to be verified.

### PR DESCRIPTION
**Pull request by zhlhahaha**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8916

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by zhlhahaha** [comment]
cc: @andreabolognani @rmohr

**Review by andreabolognani — COMMENTED** [review]
@zhlhahaha looks good save for some nits! Thanks for looking into this :)

**Comment by zhlhahaha** [comment]
@andreabolognani  PR updated.

**Comment by andreabolognani** [comment]
Everything looks fine, but I'm not exceedingly familiar with the details of the code in question so I'll leave a formal approval to someone else :)

File: pkg/virt-api/webhooks/arm64.go [hunk 1]
```diff
@@ -33,16 +33,21 @@ const (
...
-// 3. if use host-model for cpu model
-// 4. if not use 'scsi', 'virtio' as disk bus
+// ValidateVirtualMachineInstanceArm64Setting is a validation function for validating-webhook to filter unsupported setting on Arm64
 func ValidateVirtualMachineInstanceArm64Setting(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) []metav1.StatusCause {
 	var statusCauses []metav1.StatusCause
+	validateBootOptions(field, spec, &statusCauses)
+	validateCPUModel(field, spec, &statusCauses)
+	validateDiskBus(field, spec, &statusCauses)
+	validateWatchdog(field, spec, &statusCauses)
+	validateSoundDevice(field, spec, &statusCauses)
```
Comment (PiotrProkop) [inline pkg/virt-api/webhooks/arm64.go:43]: Maybe it will be more readable to change prefix from `validate` to `validateArm64*`.

Comment (zhlhahaha) [reply pkg/virt-api/webhooks/arm64.go:43]: Hi @PiotrProkop , I think twice about it. As these `validate*` functions are private function and they are in the file whose name is `arm64.go` and only called by public function `ValidateVirtualMachineInstanceArm64Setting`. These hints are enough for user understand they are for Arm64 platform. Change their name from `validate` to `validateArm64*` seems redundant.

Comment (PiotrProkop) [reply pkg/virt-api/webhooks/arm64.go:43]: Ok.

Comment (rmohr) [reply pkg/virt-api/webhooks/arm64.go:43]: Definitely not a blocker, but given that we have all functions for all architectures in one folder at the moment, the prefix would make sense. Would be a different question if we would have all arm64 validation in a seprate folder.

**Pull request merged**